tools/shared.py

```diff
@@ -169,6 +169,13 @@ def check_returncode(self):
 
 
 def run_process(cmd, check=True, input=None, *args, **kw):
+  """Runs a subpocess returning the exit code.
+
+  By default this function will raise an exception on failure.  Therefor this should only be
+  used if you want to cratch such failures.  For most subprocess failures should fatal and you
+  should use the `check_call` wrapper below instead.
+  """
```

Comment:
In this case, I think the great majority of our code uses `run_process` when it should be using `check_call` instead?

Reply:
It depends on what the call site is trying to do, but yes, we should probably audit all the run_process call sites.